pkg/cmd/server/origin/authorizer.go

```diff
@@ -51,23 +51,27 @@ func newAuthorizer(
 	podInformer coreinformers.PodInformer,
 	pvInformer coreinformers.PersistentVolumeInformer,
 	projectRequestDenyMessage string,
+
 ) (kauthorizer.Authorizer, authorizer.SubjectLocator) {
 	messageMaker := authorizer.NewForbiddenMessageResolver(projectRequestDenyMessage)
-	roleBasedAuthorizer := authorizer.NewAuthorizer(kubeAuthorizer, messageMaker)
 	subjectLocator := authorizer.NewSubjectLocator(kubeSubjectLocator)
-
-	scopeLimitedAuthorizer := scope.NewAuthorizer(roleBasedAuthorizer, clusterRoleGetter, messageMaker)
-	// Wrap with an authorizer that detects unsafe requests and modifies verbs/resources appropriately so policy can address them separately
-	browserSafeAuthorizer := browsersafe.NewBrowserSafeAuthorizer(scopeLimitedAuthorizer, user.AllAuthenticated)
+	scopeLimitedAuthorizer := scope.NewAuthorizer(clusterRoleGetter, messageMaker)
 
 	graph := node.NewGraph()
 	node.AddGraphEventHandlers(graph, podInformer, pvInformer)
 	nodeAuthorizer := node.NewAuthorizer(graph, nodeidentifier.NewDefaultNodeIdentifier(), kbootstrappolicy.NodeRules())
 
 	authorizer := authorizerunion.New(
-		authorizerfactory.NewPrivilegedGroups(user.SystemPrivilegedGroup), // authorizes system:masters to do anything, just like upstream
+		// Wrap with an authorizer that detects unsafe requests and modifies verbs/resources appropriately so policy can address them separately.
+		// Scopes are first because they will authoritatively deny and can logically be attached to anyone.
+		browsersafe.NewBrowserSafeAuthorizer(scopeLimitedAuthorizer, user.AllAuthenticated),
```

Comment:
Should scopes limit system:master ? Isn't this a change insemantics where previously system:master was omnipotent ?

Reply:
> Should scopes limit system:master ? Isn't this a change insemantics where previously system:master was omnipotent ?

If a system:master identity chooses to create a token limited by scopes, why shouldn't it be respected?  Seems more like a previous bug than a feature.

Reply:
Seems fine since `system:masters` / the node has to ask for the scopes, they are not imposed upon them somehow (since they are not OAuth based users).